src/stcal/ramp_fitting/ols_fit.py

```diff
@@ -0,0 +1,3439 @@
+
+    gain_sect : float, 2D array
+        gain values for all pixels in data section
+
+    i_max_seg : int
+        used for size of initial allocation of arrays for optional results;
+        maximum possible number of segments within the ramp, based on the
+        number of CR flags
+
+    ngroups : int
+        number of groups per integration
+
+    weighting : string
+        'optimal' specifies that optimal weighting should be used; currently
+        the only weighting supported.
+
+    f_max_seg : int
+        actual maximum number of segments within a ramp, based on the fitting
+        of all ramps; later used when truncating arrays before output.
+
+    Returns
+    -------
+    gdq_sect : int, 3D array
+        data quality flags for pixels in section
+
+    inv_var : float, 1D array
+        values of 1/variance for good pixels
+
+    opt_res : OptRes object
+        contains all quantities related to fitting for use in computing final
+        slopes, variances, etc. and is used to populate the optional output
+
+    f_max_seg : int
+        actual maximum number of segments within a ramp, updated here based on
+        fitting ramps in the current data section; later used when truncating
+        arrays before output.
+
+    num_seg : int, 1D array
+        numbers of segments for good pixels
+    """
+    ngroups, nrows, ncols = data_sect.shape
+    npix = nrows * ncols  # number of pixels in section of 2D array
+
+    all_pix = np.arange(npix)
+    arange_ngroups_col = np.arange(ngroups)[:, np.newaxis]
+    start = np.zeros(npix, dtype=np.int32)  # lowest channel in fit
+
+    # Highest channel in fit initialized to last read
+    end = np.zeros(npix, dtype=np.int32) + (ngroups - 1)
+
+    pixel_done = (end < 0)  # False until processing is done
+
+    inv_var = np.zeros(npix, dtype=np.float32)  # inverse of fit variance
+    num_seg = np.zeros(npix, dtype=np.int32)  # number of segments per pixel
+
+    # End stack array - endpoints for each pixel
+    # initialize with ngroups for each pixel; set 1st channel to 0
+    end_st = np.zeros((ngroups + 1, npix), dtype=np.int32)
+    end_st[0, :] = ngroups - 1
+
+    # end_heads is initially a tuple populated with every pixel that is
+    # either saturated or contains a cosmic ray based on the input DQ
+    # array, so is sized to accomodate the maximum possible number of
+    # pixels flagged. It is later compressed to be an array denoting
+    # the number of endpoints per pixel.
+    end_heads = np.ones(npix * ngroups, dtype=np.int32)
+
+    # Create nominal 2D ERR array, which is 1st slice of
+    #    avged_data_cube * readtime
+    err_2d_array = data_sect[0, :, :] * frame_time
+
+    # Suppress, then re-enable, harmless arithmetic warnings
+    warnings.filterwarnings("ignore", ".*invalid value.*", RuntimeWarning)
+    warnings.filterwarnings("ignore", ".*divide by zero.*", RuntimeWarning)
+    err_2d_array[err_2d_array < 0] = 0
+    warnings.resetwarnings()
```

Comment:
I commented these warnings out and all unit tests and regression tests passed.  I am not sure why these warnings are suppressed.  Looking at the operation, it is not clear it would raise either of these warnings.